test/common/redis/conn_pool_impl_test.cc

```diff
@@ -129,13 +167,44 @@ TEST_F(RedisClientImplTest, FailAll) {
   PoolRequest* handle1 = client_->makeRequest(request1, callbacks1);
   EXPECT_NE(nullptr, handle1);
 
-  EXPECT_CALL(connection_callbacks, onEvent(Network::ConnectionEvent::RemoteClose));
   EXPECT_CALL(callbacks1, onFailure());
+  EXPECT_CALL(connection_callbacks, onEvent(Network::ConnectionEvent::RemoteClose));
   upstream_connection_->raiseEvents(Network::ConnectionEvent::RemoteClose);
+
+  EXPECT_EQ(1UL, host_->cluster_.stats_.upstream_cx_destroy_with_active_rq_.value());
+  EXPECT_EQ(1UL, host_->cluster_.stats_.upstream_cx_destroy_remote_with_active_rq_.value());
+}
+
+TEST_F(RedisClientImplTest, FailAllWithCancel) {
+  InSequence s;
+
+  setup();
+  onConnected();
+
+  Network::MockConnectionCallbacks connection_callbacks;
+  client_->addConnectionCallbacks(connection_callbacks);
+
+  RespValue request1;
```

Comment:
I might clean this up in a subsequent PR but I don't think this is worth taking the time to change right now.